### PR TITLE
[android] Improve error handling when importing bookmarks

### PR DESCRIPTION
**Pull request by rtsisyk**
See #7149 #6944

### CODE REVIEW

File: data/strings/strings.txt [hunk 1]
```diff
@@ -4036,6 +4036,99 @@
...
+    sw = Aina ya faili haitambuliwi na programu:\n%1$@
+    th = แอปไม่รู้จักประเภทไฟล์:\n%1$@
+    tr = Dosya türü uygulama tarafından tanınmıyor:\n%1$@
+    uk = Тип файлу не розпізнається програмою:\n%1$@
+    vi = Loại tệp không được ứng dụng nhận dạng:\n%1$@
+    zh-Hans = 应用程序无法识别文件类型：\n%1$@
+    zh-Hant = 應用程式無法識別該文件類型：\n%1$@
+
+  [failed_to_download_file]
+    comment = Failed to download a bookmark file.
```
Comment (biodranik) [inline data/strings/strings.txt:4086]: Since when downloading of bookmarks is supported in OM?

**Comment by Jean-BaptisteC** [comment]
Why not ask users to retry import, if the problem appears always in cold start?

File: data/strings/strings.txt (hunk 1 above)
Comment (biodranik) [reply data/strings/strings.txt:4086]: Does it mean that OM can actually download bookmarks from remote URLs? Are there any proofs that its possible? AFAIU OM can open only local content URLs. Right?

Using the "Download" word now looks very misleading.

**Review by biodranik — CHANGES_REQUESTED** [review]
What exactly has been improved, and which behavior was changed?

File: data/strings/strings.txt [hunk 2]
```diff
@@ -4091,6 +4091,99 @@
     zh-Hans = 书签上传失败。该文件可能已损坏或有缺陷。
     zh-Hant = 書籤上傳失敗。 檔案可能有毀損或是不完全。
 
+  [unknown_file_type]
+    comment = Failed to recognize the format of a bookmark file.
```
Comment (biodranik) [inline data/strings/strings.txt:4095]:
```suggestion
    comment = Failed to recognize the format of a bookmarks or tracks file.
```
?

File: data/strings/strings.txt [hunk 3]
```diff
@@ -4091,6 +4091,99 @@
...
+    sw = Aina ya faili haitambuliwi na programu:\n%1$@
+    th = แอปไม่รู้จักประเภทไฟล์:\n%1$@
+    tr = Dosya türü uygulama tarafından tanınmıyor:\n%1$@
+    uk = Тип файлу не розпізнається програмою:\n%1$@
+    vi = Loại tệp không được ứng dụng nhận dạng:\n%1$@
+    zh-Hans = 应用程序无法识别文件类型：\n%1$@
+    zh-Hant = 應用程式無法識別該文件類型：\n%1$@
+
+  [failed_to_open_file]
+    comment = Failed to open a bookmark file.
```
Comment (biodranik) [inline data/strings/strings.txt:4141]:
```suggestion
    comment = Failed to open a bookmarks or tracks file in Organic Maps
```

File: android/app/src/main/java/app/organicmaps/bookmarks/data/BookmarkManager.java [hunk 4]
```diff
@@ -474,24 +477,36 @@ else if (type.equalsIgnoreCase("kml+xml"))
...
       StorageUtils.copyFile(resolver, uri, tempFile);
-    } catch (IOException e)
+      Logger.d(TAG, "Downloaded bookmarks file from " + uri + " into " + filename);
+      UiThread.run(() -> loadBookmarksFile(tempFile.getAbsolutePath(), true));
+      return true;
+    }
+    catch (IOException|SecurityException e)
     {
-      Logger.w(TAG, "Failed to download bookmarks file from " + uri, e);
+      Logger.e(TAG, "Could not download bookmarks file from " + uri, e);
```
Comment (biodranik) [inline android/app/src/main/java/app/organicmaps/bookmarks/data/BookmarkManager.java:503]: Will removed `ex.printStackTrace();` help us to decode errors in logs?

**Review by biodranik — APPROVED** [review]

**Pull request merged**